kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs

```diff
@@ -1501,21 +1528,37 @@ impl<'tcx> GotocCtx<'tcx> {
         Stmt::block(vec![expr_place, check_stmt], loc)
     }
 
-    /// simd_shuffle constructs a new vector from the elements of two input vectors,
-    /// choosing values according to an input array of indexes.
+    /// `simd_shuffle` constructs a new vector from the elements of two input
+    /// vectors, choosing values according to an input array of indexes.
+    ///
+    /// We check that:
+    ///  1. The return type length is equal to the expected length (`n`) of the
+    ///     `simd_shuffle` operation.
+    ///  2. The return type's subtype is equal to the vector's subtype (i.e.,
+    ///     the 1st argument). Both input vectors are guaranteed to be of the
+    ///     same type when they get here due to the `simd_shuffle` definition.
+    ///
+    /// In the case of some SIMD intrinsics, the backend is responsible for
+    /// performing this and similar checks, and erroring out if it proceeds.
+    ///
+    /// TODO: Check that `indexes` contains constant values which are within the
+    /// expected bounds. See
+    /// <https://github.com/model-checking/kani/issues/1960> for more details.
```

Comment:
Do you have a test that covers this case?

Reply:
We didn't, added one!